src/tiny-invariant.ts

```diff
@@ -1,13 +1,25 @@
 const isProduction: boolean = process.env.NODE_ENV === 'production';
 const prefix: string = 'Invariant failed';
 
-// Throw an error if the condition fails
-// Strip out error messages for production
-// > Not providing an inline default argument for message as the result is smaller
+/**
+ * __invariant__
+ * 
+ * The `invariant` function takes a value, and if the value is falsy then the `invariant` function will throw. If the value is truthy, then the function will not throw.
+ *
+ * @param {boolean} condition A boolean condition - if falsey will thrown an error.
+ * @param {(string|() => string)} message The message provided to accompany the invariant. Can provide a string, or a function that returns a string for cases where the message takes a fair amount of effort to compute.
+ *  
+ * @returns {asserts condition is true}
```

Comment:
asserts condition is truthy?

Reply:
`{ }` syntax is a hint for the TS server - so this is really
`asserts condition (variable) is true` <-- which is the ts expression

Can certainly add the explanation as an annotation tho next to it.